runtime/turing/src/migrations.rs

```diff
@@ -0,0 +1,526 @@
+use super::*;
+use frame_support::traits::OnRuntimeUpgrade;
+
+pub mod assets {
+	use super::*;
+	use frame_support::{Blake2_128Concat, BoundedVec, Twox64Concat, WeakBoundedVec};
+	use orml_asset_registry::AssetMetadata;
+
+	pub type AssetMetadataOf = AssetMetadata<Balance, CustomMetadata>;
+
+	pub mod parachains {
+		pub mod testchain {
+			pub const ID: u32 = 1999;
+		}
+
+		pub mod heiko {
+			pub const ID: u32 = 2085;
+			pub const HKO_KEY: &[u8] = b"HKO";
+			pub const SKSM_KEY: &[u8] = b"sKSM";
+		}
+
+		pub mod karura {
+			pub const ID: u32 = 2000;
+			pub const KAR_KEY: &[u8] = &[0, 128];
+			pub const AUSD_KEY: &[u8] = &[0, 129];
+			pub const LKSM_KEY: &[u8] = &[0, 131];
+		}
+
+		pub mod khala {
+			pub const ID: u32 = 2004;
+		}
+	}
+
+	pub fn dollar(decimals: u32) -> Balance {
+		10_u128.pow(decimals)
+	}
+
+	pub fn cent(decimals: u32) -> Balance {
+		dollar(decimals) / 100
+	}
+
+	pub fn millicent(decimals: u32) -> Balance {
+		cent(decimals) / 1_000
+	}
+
+	/// Based on the precedent set by other projects. This will need to be changed.
+	pub fn ksm_per_second() -> u128 {
+		cent(12) * 16
+	}
+
+	/// Assuming ~ $0.50 TUR price.
+	pub fn tur_per_second() -> u128 {
+		let ksm_decimals: u32 = 12;
+		let tur_decimals: u32 = 10;
+		let diff = ksm_decimals - tur_decimals;
+		let tur_equivalent = ksm_per_second() / 10_u128.pow(diff);
+
+		// Assuming KSM ~ $130.00.
+		tur_equivalent * 260
+	}
+
+	#[frame_support::storage_alias]
+	type TotalIssuance = StorageMap<Tokens, Twox64Concat, CurrencyId, Balance>;
+
+	#[frame_support::storage_alias]
+	type Locks<T: Config> = StorageDoubleMap<
+		Tokens,
+		Blake2_128Concat,
+		AccountId,
+		Twox64Concat,
+		CurrencyId,
+		BoundedVec<orml_tokens::BalanceLock<Balance>, <T as orml_tokens::Config>::MaxLocks>,
+	>;
+
+	#[frame_support::storage_alias]
+	type Accounts = StorageDoubleMap<
+		Tokens,
+		Blake2_128Concat,
+		AccountId,
+		Twox64Concat,
+		CurrencyId,
+		orml_tokens::AccountData<Balance>,
+	>;
+
+	#[frame_support::storage_alias]
+	type Reserves<T: Config> = StorageDoubleMap<
+		Tokens,
+		Blake2_128Concat,
+		AccountId,
+		Twox64Concat,
+		CurrencyId,
+		BoundedVec<
+			orml_tokens::ReserveData<<T as orml_tokens::Config>::ReserveIdentifier, Balance>,
+			<T as orml_tokens::Config>::MaxReserves,
+		>,
+	>;
+
+	#[frame_support::storage_alias]
+	type LastAssetId<T: Config> =
+		StorageValue<AssetRegistry, <T as orml_asset_registry::Config>::AssetId>;
+
+	pub struct AssetRegistryMigration;
+	impl OnRuntimeUpgrade for AssetRegistryMigration {
+		fn on_runtime_upgrade() -> Weight {
+			log::info!(
+				target: "asset_registry",
+				"on_runtime_upgrade: Attempted to apply asset_registry migration"
+			);
+
+			let assets = vec![
```

Comment:
What do you think about moving this vec outside of `on_runtime_upgrade`? It would make this method easier to read and you could assert on `vec.len()` in the post_upgrade.